test/functional/importer_controller_test.rb

```diff
@@ -0,0 +1,130 @@
+require File.expand_path('../../test_helper', __FILE__)
+
+class ImporterControllerTest < ActionController::TestCase
+  def setup
+    @project = Project.create! :name => 'foo'
+    @tracker = @project.trackers.create(:name => 'Defect')
+    @role = Role.create! :name => 'ADMIN', :permissions => [:import]
+    @user = create_user!(@role, @project)
+    @iip = create_iip_for_multivalues!(@user, @project)
+    @issue = create_issue!(@project, @user)
+    create_custom_fields!(@issue)
+    create_versions!(@project)
+    User.stubs(:current).returns(@user)
+  end
+
+  test 'should handle multi-value fields' do
+    post :result, build_params
+    assert_response :success
+    @issue.reload
+    assert_issue_has_affected_versions(@issue, ['Admin', '2013-09-25'])
+  end
+
+  test 'should handle single-value fields' do
+    assert_equal 'foobar', @issue.subject
+    post :result, build_params
+    assert_response :success
+    @issue.reload
+    assert_equal 'barfooz', @issue.subject
+  end
+
+  test 'should create issue if none exists' do
+    Issue.delete_all
+    assert_equal 0, Issue.count
+    post :result, build_params(:update_issue => nil)
+    assert_response :success
+    assert_equal 1, Issue.count
+    issue = Issue.first
+    assert_equal 'barfooz', issue.subject
+  end
+
+  protected
+
+  def build_params(opts={})
+    opts.reverse_merge(
+      :import_timestamp => @iip.created.strftime("%Y-%m-%d %H:%M:%S"),
+      :update_issue => 'true',
+      :unique_field => '#',
+      :project_id => @project.id,
+      :fields_map => {
+        '#' => 'id',
+        'Subject' => 'subject',
+        'Tags' => 'Tags',
+        'Affected versions' => 'Affected versions',
+        'Priority' => 'priority',
+        'Tracker' => 'tracker',
+        'Status' => 'status'
+      }
+    )
+  end
+
+  def assert_issue_has_affected_versions(issue, version_names)
+    version_ids = version_names.map do |name|
+      Version.find_by_name!(name).id.to_s
+    end
+    versions_field = CustomField.find_by_name! 'Affected versions'
+    values = issue.custom_values.find_all_by_custom_field_id versions_field.id
+    assert values.any? {|v| version_ids.include?(v.value) }
+  end
+
+  def create_user!(role, project)
+    user = User.new :admin => true,
+                     :firstname => 'Bob',
+                     :lastname => 'Loblaw',
+                     :mail => 'bob.loblaw@law.blog'
```

Comment:
note on style, I find it BEST PRACTIVE to always use @example.com - you never know when your LOL domain might actually be taken.